cpp/wsWebsocketConnection.h

```diff
@@ -348,28 +367,33 @@ class wsWebsocketConnection
       return false;
     }
 
-    // Read a message into our buffer
-    // This buffer will hold the incoming message
-    beast::multi_buffer buffer;
-    ws.read(buffer, ec);
-    if(ec)
-    {
-      this->HandleError("read", ec);
-      return false;
-    }
-
     if (result)
     {
-      *result = json::parse(beast::buffers_to_string(buffer.data()));
-    }
+      std::unique_lock<std::mutex> lock (MutexRPCMessages);
+      while (QueueRPCMessages.empty()) {
+        CVRPCMessages.wait(lock);
+      }
 
-    // The make_printable() function helps print a ConstBufferSequence
-    if (this->Debug)
-    {
-      std::cout << beast::buffers_to_string(buffer.data()) << std::endl;
+      *result = QueueRPCMessages.front();
+      QueueRPCMessages.pop();
```

Comment:
Why do you keep the front and remove the last?

Reply:
Unlike the `std::list` or `std::stack`, the `std::queue::pop` deletes the first element.